### PR TITLE
Simple Hash is removed.

### PR DESCRIPTION
**Pull request by eccles**
The proof mechanism is now not specified anywhere and accepts and uses whatever the backend supples.

The customer can still specify the proof mechanism as MERKLE_LOG but
cannot specify any other value.

[AB#9189](https://dev.azure.com/jitsuin/0629f48c-3979-4bbc-9026-cb06b3dfd0ae/_workitems/edit/9189)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -85,21 +84,13 @@ You can then use the examples code to create assets (see examples directory):
...
-            "proof_mechanism": ProofMechanism.SIMPLE_HASH.name,
-        }
-
-        # The first argument are the properties of the asset
-        # The second argument are the attributes of the asset
-        # The third argument is wait for confirmation:
+        # The first argument are the attributes of the asset
+        # The second argument is wait for confirmation:
         #   If @confirm@ is True then this function will not
         #   return until the asset is confirmed on the blockchain and ready
```
Comment (honourfish) [inline README.rst:90]: minor: driveby comment maybe we should remove mention of blockchain?

its not true for simplehash or merklelog assets

File: archivist/proof_mechanism.py [hunk 2]
```diff
@@ -14,8 +16,7 @@ class ProofMechanism(Enum):
     """Enumerate proof mechanism options"""
 
     # previously used but now removed
-    __RESERVED = 1
```
Comment (honourfish) [inline archivist/proof_mechanism.py:17]: proof mechs dropping like flies!

**Review by honourfish — APPROVED** [review]

File: README.rst (hunk 1 above)
Comment (eccles) [reply README.rst:90]: nice spot - blockchain mentioned elsewhere as well

File: archivist/proof_mechanism.py (hunk 2 above)
Comment (eccles) [reply archivist/proof_mechanism.py:17]: yay...

**Review by honourfish — APPROVED** [review]
